kf_lib_data_ingest/common/warehousing.py

```diff
@@ -0,0 +1,72 @@
+from urllib.parse import urlparse
+
+import sqlalchemy
+
+
+def db_study_url(db_maintenance_url, study_id):
+    return urlparse(db_maintenance_url)._replace(path=f"/{study_id}").geturl()
+
+
+def init_study_db(db_maintenance_url, study_id):
+    """
+    Create and/or reset a warehouse database for the given study.
+
+    :param db_maintenance_url: Connection URL with login/password for the warehouse db
+    :type db_maintenance_url: str
+    :param study_id: Unique identifier for the study
+    :type study_id: str
+    """
+    # create the study database if needed
+    eng = sqlalchemy.create_engine(
+        db_maintenance_url, isolation_level="AUTOCOMMIT"
+    )
+    try:
+        eng.execute(f'CREATE DATABASE "{study_id}"')
+    except sqlalchemy.exc.ProgrammingError as e:
+        if "already exists" not in str(e):
+            raise
+    eng.dispose()
+
+    # link to the study database
+    eng = sqlalchemy.create_engine(db_study_url(db_maintenance_url, study_id))
+
+    # reset study database
+    eng.drop_all()
+    eng.dispose()
+
```

Comment:
Can you add a log statement here: 
```
logger.info(
        f'Created study db: {study_id} in warehouse {the obfuscated url here}'
    )
```

Reply:
FFFFF. I've fixed this a dozen times and it keeps getting reverted somehow.